activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -43,65 +50,181 @@ class InvalidMessage < StandardError; end
     # * <tt>:digest</tt> - String of digest to use for signing. Default is +SHA1+.
     # * <tt>:serializer</tt> - Object serializer to use. Default is +Marshal+.
     def initialize(secret, *signature_key_or_options)
-      options = signature_key_or_options.extract_options!
-      sign_secret = signature_key_or_options.first
+      @options = signature_key_or_options.extract_options!
```

Comment:
This is as it was implemented before. They probably did it because we don't want passing signature key to be mandatory. I am not sure if we could pass signature key using **options. What do you say ?